app/server/models/User.js

```diff
@@ -316,6 +332,21 @@ schema.methods.generateAuthToken = function(){
   return jwt.sign(this._id, JWT_SECRET);
 };
 
+// Generate temp login code
+schema.methods.generateTempCode = function() {
+  // TOOD: way to generate a random code securely?
```

Comment:
Because this is acting like a password, we probably want it to be cryptographically secure. One course we could take is adding a node package to do that, but I'm not sure if that's the best/only path.